src/Evolution/Systems/GrMhd/GhValenciaDivClean/TimeDerivativeTerms.hpp

```diff
@@ -0,0 +1,212 @@
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/TimeDerivative.hpp"
+#include "Evolution/Systems/GrMhd/GhValenciaDivClean/StressEnergy.hpp"
+#include "Evolution/Systems/GrMhd/GhValenciaDivClean/Tags.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/TimeDerivativeTerms.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Literals.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace grmhd::GhValenciaDivClean {
+namespace detail {
+
+// prefix tag used to disambiguate tag collisions
+template <typename Tag>
+struct ValenciaTempTag : db::SimpleTag, db::PrefixTag {
+  using type = typename Tag::type;
+};
+
+template <typename Tag>
+struct TemporaryReference {
+  using tag = Tag;
+  using type = const typename Tag::type&;
+};
+
+template <typename GhDtTagList, typename ValenciaDtTagList,
+          typename ValenciaFluxTagList, typename GhTempTagList,
+          typename ValenciaTempTagList, typename GhGradientTagList,
+          typename GhArgTagList, typename ValenciaArgTagList,
+          typename ValenciaTimeDerivativeArgTagList>
+struct TimeDerivativeTermsImpl;
+
+template <typename... GhDtTags, typename... ValenciaDtTags,
+          typename... ValenciaFluxTags, typename... GhTempTags,
+          typename... ValenciaTempTags, typename... GhGradientTags,
+          typename... GhArgTags, typename... ValenciaArgTags,
+          typename... ValenciaTimeDerivativeArgTags>
+struct TimeDerivativeTermsImpl<
+    tmpl::list<GhDtTags...>, tmpl::list<ValenciaDtTags...>,
+    tmpl::list<ValenciaFluxTags...>, tmpl::list<GhTempTags...>,
+    tmpl::list<ValenciaTempTags...>, tmpl::list<GhGradientTags...>,
+    tmpl::list<GhArgTags...>, tmpl::list<ValenciaArgTags...>,
+    tmpl::list<ValenciaTimeDerivativeArgTags...>> {
+  static void apply(
+      const gsl::not_null<typename GhDtTags::type*>... gh_dts,
+      const gsl::not_null<typename ValenciaDtTags::type*>... valencia_dts,
+      const gsl::not_null<typename ValenciaFluxTags::type*>... valencia_fluxes,
+      const gsl::not_null<typename GhTempTags::type*>... gh_temporaries,
+      const gsl::not_null<
+          typename ValenciaTempTags::type*>... valencia_temporaries,
+      gsl::not_null<tnsr::aa<DataVector, 3_st>*> stress_energy,
+      gsl::not_null<tnsr::a<DataVector, 3_st>*> four_velocity_one_form,
+      gsl::not_null<tnsr::a<DataVector, 3_st>*>
+          comoving_magnetic_field_one_form,
+      const typename ::Tags::deriv<GhGradientTags, tmpl::size_t<3_st>,
+                                   Frame::Inertial>::type&... gh_gradients,
+      const typename GhArgTags::type&... gh_args,
+      const typename ValenciaArgTags::type&... valencia_args) noexcept {
+    GeneralizedHarmonic::TimeDerivative<3_st>::apply(
+        gh_dts..., gh_temporaries..., gh_gradients..., gh_args...);
+
+    // used for reordering the packs
```

Comment:
I agree that this is what is being done, could you add a brief sentence as to why? Maybe something like `This is needed to be able to reuse temporary tags that the GH system computed already and pass them in as argument tags to the GRMHD system`?